x-pack/plugins/ml/public/application/components/feedback_button/feedback_button.tsx

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useState, useEffect, FC, useMemo, useRef } from 'react';
+import { MlPages, ML_PAGES } from '../../../locator';
+import { useMlKibana } from '../../contexts/kibana';
+import { useEnabledFeatures } from '../../contexts/ml';
+
+import { useJobsApiService } from '../../services/ml_api_service/jobs';
+import { useCloudCheck } from '../node_available_warning/hooks';
+import { FeatureFeedbackButton } from './feature_feedback_button';
+
+interface Props {
+  jobIds: string[];
+  page: MlPages;
+}
+
+const FORM_IDS = {
+  SINGLE_METRIC_VIEWER: '1FAIpQLSdlMYe3wuJh2KtBLajI4EVoUljAhGjJwjZI7zUY_Kn_Sr2lug',
+  ANOMALY_EXPLORER: '1FAIpQLSfF1Ry561b4lYrY7iiyXhuZpxFzAmy2c9BFUT3J2AJUevY1iw',
+};
+
+const MATCHED_CREATED_BY_TAGS = ['ml-module-metrics-ui-hosts'];
+
+export const FeedBackButton: FC<Props> = ({ jobIds, page }) => {
+  const { jobs: getJobs } = useJobsApiService();
+  const {
+    services: { kibanaVersion },
+  } = useMlKibana();
+  const { isCloud } = useCloudCheck();
+  // ML does not have an explicit isServerless flag,
+  // it does however have individual feature flags which are set depending
+  // whether the environment is serverless or not.
+  // showNodeInfo will always be false in a serverless environment
+  // and true in a non-serverless environment.
+  const { showNodeInfo } = useEnabledFeatures();
+
+  const [jobIdsString, setJobIdsString] = useState<string | null>(null);
+  const [showButton, setShowButton] = useState(false);
+
+  const formId = useMemo(() => getFormId(page), [page]);
+  const mounted = useRef(false);
```

Comment:
nit: you can use https://github.com/streamich/react-use/blob/master/docs/useMountedState.md instead

Reply:
TIL
That's much nicer, updated in https://github.com/elastic/kibana/pull/175613/commits/bb3864413981a2c5b37e1a5c749fb607e03a1a6a